### ISSUE TITLE
Python 3 bindings

### ISSUE THREAD

**Issue by vshymanskyy** [wasm3/wasm3#90]
Implement `Python 3` bindings, so `wasm3` can be easily used from Python.

**Comment by abalkin** [comment]
Is it currently possible to compile wasm3 into a C library? If yes then how?

**Comment by vshymanskyy** [comment]
@abalkin you're probably asking the same question as in #116 

However I think for **Python**, the regular `distutils`-based pipeline should be used: https://docs.python.org/3/extending/building.html

**Comment by abalkin** [comment]
@vshymanskyy - right - just listing the source files in setup.py is a workable approach.

**Comment by abalkin** [comment]
On the other hand, platforms/cpp contains cmake logic to build libm3 library.

**Comment by prabirshrestha** [comment]
Also if you are using pure make and clang you can use this to build. Just make sure to update source correctly.

https://gist.github.com/prabirshrestha/5c1712ba958bb10e1d7aef99d244fc0a

**Comment by abalkin** [comment]
@vshymanskyy - I've created a draft PR (see #120) to give other developers a chance to comment on the design choices early on.  I am trying to closely follow the class structure in the existing C++ bindings.

A quick question: where should I put tests: under the top level test directory or somewhere under platforms/python?

**Comment by vshymanskyy** [comment]
Thanks @abalkin, this looks really good. I'd say `platforms/python` is a better location for tests.

**Comment by arrisde** [comment]
A while ago I looked into creating a [CFFI](https://cffi.readthedocs.io) wrapper for wasm3. With some fiddling I got it to work eventually, but the Python interface provided by CFFI is not very pleasant to use. I meant to write some wrapper classes to make this more usable but now seeing #120, which looks much cleaner to me, it may not be worth the effort anymore. I put the code [here](https://github.com/arrisde/pywasm3) in case anyone wants to check it out. 

Anyways, I think an alternative to the `distutils` approach and listing all source files in `setup.py` could be to use something like [scikit-build](https://scikit-build.readthedocs.io) which extends `setuptools` to work with CMake. I think this would integrate nicely with the existing CMake pipeline of wasm3 and could easily produce binary wheels for any CMake supported platform. Any thoughts on this?
